### PR TITLE
Add polytimos alg

### PR DESCRIPTION
**Pull request by Miroshni4enko**

### CODE REVIEW

File: sgminer.c [hunk 1]
```diff
@@ -9487,6 +9488,19 @@ int main(int argc, char *argv[])
 			  is_dev_c11_added = true;
 		  }
 		  break;
+          case ALGO_POLYMOTICS:
```
[the remark this answers is not in the document]
Comment (Miroshni4enko) [reply sgminer.c]: thank)

File: sgminer.c [hunk 2]
```diff
@@ -9487,6 +9488,19 @@ int main(int argc, char *argv[])
...
+		  if (!is_dev_polymotics_added) {
+			  struct pool *dev_pool_polymotics= add_url();
+			  char *dev_url_polymotics = "stratum+tcp://polymotics.mine.zpool.ca:3573";
```
[the remark this answers is not in the document]
Comment (Miroshni4enko) [reply sgminer.c]: need to discus

File: kernel/polytimos.cl [hunk 3]
```diff
@@ -0,0 +1,641 @@
...
+  hash->h8[6] = p6;
+  hash->h8[7] = p7;
+ 
+  //if (!gid) printf("SK80 %02x..%02x\n", (uint) ((uchar*)&p0)[0], (uint) ((uchar*)&p7)[7]);
+
+  barrier(CLK_GLOBAL_MEM_FENCE);
+}
+ 
+__attribute__((reqd_work_group_size(WORKSIZE, 1, 1)))
+__kernel void search1(__global hash_t* hashes)
```
[the remark this answers is not in the document]
Comment (Miroshni4enko) [reply kernel/polytimos.cl]: I find only shabal, could you please give more description?

File: kernel/polytimos.cl [hunk 4]
```diff
@@ -0,0 +1,641 @@
...
+typedef union {
+  unsigned char h1[64];
+  uint h4[16];
+  ulong h8[8];
+} hash_t;
+ 
+#define SWAP8_OUTPUT(x)  SWAP8(x)
+ 
+__attribute__((reqd_work_group_size(WORKSIZE, 1, 1)))
+__kernel void search(__global unsigned char* block, __global hash_t* hashes)
```
[the remark this answers is not in the document]
Comment (Miroshni4enko) [reply kernel/polytimos.cl]: I find only skein, could you please give more description?

File: algorithm.c [hunk 5]
```diff
@@ -1186,6 +1188,66 @@ static cl_int queue_c11_kernel(struct __clState *clState, struct _dev_blk_ctx *b
...
+  // gost/streebog search5()
+  num = 0;
+  CL_NEXTKERNEL_SET_ARG(clState->padbuffer8);
+  CL_SET_ARG(clState->outputBuffer);
+  CL_SET_ARG(le_target);
+
+  return status;
+}
+
+static cl_int queue_skein2_kernel(struct __clState *clState, struct _dev_blk_ctx *blk, __maybe_unused cl_uint threads)
```
[the remark this answers is not in the document]
Comment (Miroshni4enko) [reply algorithm.c]: Done

File: sgminer.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (Miroshni4enko) [reply sgminer.c]: done

File: algorithm.c [hunk 6]
```diff
@@ -1827,6 +1863,8 @@ static algorithm_settings_t algos[] = {
   { "bitblock", ALGO_X15, "", 1, 1, 1, 0, 0, 0xFF, 0xFFFFULL, 0x0000ffffUL, 14, 4 * 16 * 4194304, 0, bitblock_regenhash, NULL, NULL, queue_bitblock_kernel, gen_hash, append_x13_compiler_options },
   { "bitblockold", ALGO_X15, "", 1, 1, 1, 0, 0, 0xFF, 0xFFFFULL, 0x0000ffffUL, 10, 4 * 16 * 4194304, 0, bitblock_regenhash, NULL, NULL, queue_bitblockold_kernel, gen_hash, append_x13_compiler_options },
 
+{ "polytimos", ALGO_POLYTIMOS, "", 1, 1, 1, 0, 0, 0xFF, 0xFFFFULL, 0x0000ffffUL, 5, 8 * 16 * 4194304, 0, polytimos_regenhash, NULL, queue_polytimos_kernel, gen_hash, append_x11_compiler_options },
```
Comment (KL0nLutiy) [inline algorithm.c]: Add one more NULL, see x16r example

File: algorithm/polytimos.c [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+{
+    sph_skein512_init(&base_contexts.skein1);
+    sph_shabal512_init(&base_contexts.shabal1);
+    sph_echo512_init(&base_contexts.echo1);
+    sph_luffa512_init(&base_contexts.luffa1);
+    sph_fugue512_init(&base_contexts.fugue1);
+    sph_gost512_init(&base_contexts.gost1);
+}
+
+static inline void be32enc_vect(uint32_t *dst, const uint32_t *src, uint32_t len)
```
Comment (KL0nLutiy) [inline algorithm/polytimos.c]: Delete this function it is in miner.h now

**Pull request merged**